hexrdgui/mask_manager.py

```diff
@@ -0,0 +1,281 @@
+import math
+from hexrdgui.constants import ViewType
+from hexrdgui.create_polar_mask import (
+    convert_raw_to_polar, create_polar_mask_from_raw
+)
+from hexrdgui.create_raw_mask import (
+    apply_threshold_mask, convert_polar_to_raw, create_raw_mask
+)
+from hexrdgui.hexrd_config import HexrdConfig
+from hexrdgui.mask_compatability import load_old_mask_file
+from hexrdgui.singletons import Singleton
+from hexrdgui.utils import unique_name
+
+from abc import ABC, abstractmethod
+
+
+class Mask(ABC):
+    def __init__(self, mtype='', name='', mask_image=True, show_border=False):
+        self._mask_type = mtype
+        self._name = name
+        self._mask_image = mask_image
+        self._show_border = show_border
+        self._masked_arrays = None
+
+    @property
+    def masked_arrays(self):
+        return self._masked_arrays
+
+    @masked_arrays.setter
+    def masked_arrays(self, arrs):
+        self._masked_arrays = arrs
+
+    @property
+    def mask_type(self):
+        return self._mask_type
+
+    @mask_type.setter
+    def mask_type(self, mtype):
+        self._mask_type = mtype
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, new_name):
+        self._name = new_name
+
+    @property
+    def mask_image(self):
+        return self._mask_image
+
+    @mask_image.setter
+    def mask_image(self, status):
+        self._mask_image = status
+
+    @property
+    def show_border(self):
+        return self._show_border
+
+    @show_border.setter
+    def show_border(self, status):
+        self._show_border = status
+
+    # Abstract methods
+    @abstractmethod
+    def get_data(self):
+        pass
+
+    @abstractmethod
+    def set_data(self, data):
+        pass
+
+    @abstractmethod
+    def get_mask_arrays(self):
```

Comment:
What's the difference between `get_data` and `get_mask_arrays`?

Reply:
`get_data` returns the actual point data describing the region to mask (needed for mask boundaries) and `get_mask_arrays` returns the list of masked images (one for each mask so that they can be quickly combined and re-combined if visibility is toggled).

Reply:
After some further simplification `get_mask_arrays` is now just an attribute: `masked_arrays`.